pkg/controller/upgradeconfig/upgradeconfig_controller.go

```diff
@@ -261,6 +271,13 @@ func (r *ReconcileUpgradeConfig) Reconcile(request reconcile.Request) (reconcile
 		reqLogger.Info("Cluster is already upgraded")
 		return reconcile.Result{}, nil
 	case upgradev1alpha1.UpgradePhaseFailed:
+		instance.Status.History[0].WindowBreached = true
```

Comment:
`[0]` makes sense here but its best to retrieve the history based on the version

Reply:
+1 to this. All lookups in History should probably be done with `GetHistory(<version>)` and a `nil` check performed afterwards to be safe, in the interests of accidental panic prevention. 

If that starts to make the code look ugly and excessively verbose, perhaps it could use wrapping in some function that performs the required updates safely.